backend/clubs/serializers.py

```diff
@@ -1819,6 +1820,51 @@ class Meta:
         fields = ("id", "name")
 
 
+class ClubApplicationSerializer(serializers.ModelSerializer):
+    club = serializers.SlugRelatedField(queryset=Club.objects.all(), slug_field="code")
+    name = serializers.SerializerMethodField("get_name")
+
+    def get_name(self, obj):
+        if obj.name:
+            return obj.name
+        return "{}-{}".format(obj.club.name, obj.application_start_time.strftime("%b %d, %Y"))
```

Comment:
It would be good to format this a little nicer! It'll be displayed directly to the user without any modification from the frontend.

I would recommend something like `f"{obj.club.name} Application"` and leave out the date, as that is a pretty generic default that should work for most cases. We'll render the application start and end times in the frontend anyways and make it look nicer, so no need to duplicate that information.

Also check out `f"{...}"` strings, they're a shortcut for `"{}".format(...)`! They're relatively new (Python 3.6+), but we've upgraded everything enough to use them now.